airflow/providers/openlineage/plugins/listener.py

```diff
@@ -250,7 +250,7 @@ def on_failure():
     @property
     def executor(self):
         if not self._executor:
-            self._executor = ThreadPoolExecutor(max_workers=8, thread_name_prefix="openlineage_")
+            self._executor = ProcessPoolExecutor(max_workers=8)
```

Comment:
I guess it is too optimistic to create a new 8 processes, even regular airflow components use less processes by default.
It is not cheap and in additional when we use new processes we need to be sure that arguments which pass to new processes could be serialised by `pickle`, that is also not cheap and potentially might cause some vulnerability.

Reply:
You're right. I left the same number by default which wasn't best decision.
I reduced it to 1 process, tested with 1000 DagRuns concurrent and still no choke.

Reply:
What change did this make to the memory usage when running?

Reply:
With the same scenario of 1k DagRuns, locally and in the cloud: memory consumption in scheduler increased ~100 MB.

Reply:
Maybe make it configurable? https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/configurations-ref.html

Reply:
You're reading in my mind. I was about to add it 🙂